core/txpool/txpool.go

```diff
@@ -1028,7 +1092,7 @@ func (pool *TxPool) removeTx(hash common.Hash, outofbound bool) {
 			pool.pendingNonces.setIfLower(addr, tx.Nonce())
 			// Reduce the pending counter
 			pendingGauge.Dec(int64(1 + len(invalids)))
-			return
+			return len(invalids)
```

Comment:
Your docstring change says it returns the number of txs moved to future. 
In htat case, it's correct what you return here. 
OTOH, in one place you do `dropped := pool.removeTx(tx.Hash(), false)`. If that's what you want, then you should change the docstring and instead return this, no? 

```suggestion
			return 1+ len(invalids)
```
Judging by the line above. 

And then you'd also have to change the last `return 0 ` into `return 1`

Reply:
Now the docstring was made a liar.